app/routes/projects/unarchive.tsx

```diff
@@ -0,0 +1,21 @@
+import type { ActionFunction } from "@remix-run/node";
+import { redirect } from "@remix-run/node";
+import { requireProfile, requireUser } from "~/session.server";
+import { unarchiveProject } from "~/models/project.server";
+import { adminRoleName } from "~/constants";
+
+export const action: ActionFunction = async ({ request }) => {
+  let formData = await request.formData();
+  let projectId: string = formData.get("projectId") as string;
+  const profile = await requireProfile(request);
+  const user = await requireUser(request);
+  const isAdmin = user.role == adminRoleName;
+
+  try {
+    await unarchiveProject(projectId, profile.id, isAdmin);
```

Comment:
Since both archive and unarchive routes are pretty similar, I wonder if we should instead have a single `/toggle-archive` route. 🙈 . But this change is not required, is just a nitpick, if you like it as it is, you can ignore it.